src/stactools/core/io/__init__.py

```diff
@@ -22,13 +22,13 @@ def read_text(href: str,
 class FsspecStacIO(DefaultStacIO):
     def read_text_from_href(self, href: str, *args: Any, **kwargs: Any) -> str:
         with fsspec.open(href, "r") as f:
-            return f.read()
+            return str(f.read())
```

Comment:
```suggestion
            return str(f.read(), encoding='utf-8')
```

Ref: https://docs.python.org/3/library/stdtypes.html#str

Reply:
Forcing the UTF8 decoding was causing tests to fail, so I did something similar to the XML handling change you suggested below.